src/secret/ledger_nano.rs

```diff
@@ -101,154 +129,160 @@ impl SecretManageExt for LedgerSecretManager {
         // lock the mutex to prevent multiple simultaneous requests to a ledger
         let _lock = self.mutex.lock().await;
 
-        // Get coin type and account index from first input
-        let (_coin_type, account_index) = match &prepared_transaction.inputs_data.first() {
-            Some(input) => {
-                match &input.chain {
-                    Some(chain) => {
-                        let raw: Vec<u32> = chain
-                            .segments()
-                            .iter()
-                            // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
-                            .map(|seg| u32::from_be_bytes(seg.bs()))
-                            .collect();
-                        (raw[0], raw[1])
+        let mut input_bip32_indices: Vec<iota_ledger::LedgerBIP32Index> = Vec::new();
+        let mut coin_type: Option<u32> = None;
+        let mut account_index: Option<u32> = None;
+
+        let input_len = prepared_transaction.inputs_data.len();
+
+        for input in &prepared_transaction.inputs_data {
+            let bip32_indices: Vec<u32> = match &input.chain {
+                Some(chain) => {
+                    chain
+                        .segments()
+                        .iter()
+                        // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
+                        .map(|seg| u32::from_be_bytes(seg.bs()))
+                        .collect()
```

Comment:
This isn't needed, if we can update crypto.rs in ledger.rs to the same version

Reply:
how would the alternative look like if we had the same version?